src/test/java/uk/gov/pay/api/json/AuthorisationAPIRequestDeserializerTest.java

```diff
@@ -0,0 +1,191 @@
+package uk.gov.pay.api.json;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import uk.gov.pay.api.exception.BadAuthorisationAPIRequestException;
+import uk.gov.pay.api.model.AuthorisationAPIRequest;
+import uk.gov.pay.api.model.PaymentError;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThrows;
+
+class AuthorisationAPIRequestDeserializerTest {
```

Comment:
Would be good to parameterise these tests so different criteria (blank, missing, length, type) can be tested without repeating similar setup/assertions across.